src/xo-app/settings/ips/index.js

```diff
@@ -0,0 +1,329 @@
+            </Col>
+            <Col mediumSize={1} offset={6}>
+              <ActionRowButton
+                handler={this._deleteIp}
+                handlerParam={ip}
+                icon='delete'
+              />
+            </Col>
+          </Row>
+        }
+        const addressVifs = ipPool.addresses[ip].vifs
+        return <Row>
+          <Col mediumSize={5}>
+            <strong>{ip}</strong>
+          </Col>
+          <Col mediumSize={6}>{!isEmpty(addressVifs)
+            ? map(addressVifs, vifId => {
+              const vif = vifs[vifId][0]
+              const network = networks[vif.$network][0]
+
+              return `${network.name_label} #${vif.device}`
+            }).join(', ')
+            : <em>{_('ipsNotUsed')}</em>
+          }</Col>
+          <Col mediumSize={1}>
+            <ActionRowButton
+              handler={this._deleteIp}
+              handlerParam={ip}
+              icon='delete'
+            />
+          </Col>
+        </Row>
+      })}
+      <Row>
+        <Col>
+          {showNewIpForm
+          ? <form id='newIpForm' className='form-inline'>
+            <ActionButton btnStyle='danger' handler={this._toggleNewIps} icon='remove' />
+            {' '}
+            <DebounceInput
+              autoFocus
+              onChange={event => this._onChangeNewIps(event.target.value, ipPool.id)}
+              type='text'
+              className='form-control'
+              required
+              value={newIps || ''}
+            />
+            {' '}
+            <ActionButton form={`newIpForm`} icon='save' btnStyle='primary' handler={this._addIps} />
+          </form>
+          : <ActionButton btnStyle='success' size='small' handler={this._toggleNewIps} icon='add' />}
+        </Col>
+      </Row>
+    </Container>
+  }
+}
+
+@connectStore(() => {
+  const getNetworks = createGetObjectsOfType('network')
+
+  return (state, props) => ({
+    networks: getNetworks(state, props)
+  })
+})
+class NetworksCell extends BaseComponent {
+  _addNetworks = () => {
+    if (isEmpty(this.state.newNetworks)) {
+      return this._toggleNewNetworks()
+    }
+    const { ipPool } = this.props
+    setIpPool(ipPool.id, {
+      networks: [ ...ipPool.networks, ...this.state.newNetworks ]
+    })
+    this._toggleNewNetworks()
+    this.setState({ newNetworks: undefined })
+  }
+
+  _deleteNetwork = networkId => {
+    const _networks = [ ...this.props.ipPool.networks ]
+    const index = findIndex(_networks, network => network === networkId)
+    if (index !== -1) {
+      _networks.splice(index, 1)
+      setIpPool(this.props.ipPool.id, { networks: _networks })
+    }
+  }
+
+  _toggleNewNetworks = () =>
+    this.setState({ showNewNetworkForm: !this.state.showNewNetworkForm })
+  _onChangeNewNetworks = newNetworks =>
+    this.setState({ newNetworks: map(newNetworks, network => network.id) })
+  _networkPredicate = network =>
+    !includes(this.props.ipPool.networks, network.id)
```

Comment:
If the predicate has some dependencies they should be explicit by regenerating the predicate `createSelector()`.